hack/lib/catalogsource.bash

```diff
@@ -1,22 +1,101 @@
 #!/usr/bin/env bash
 
-function ensure_catalogsource_installed {
-  logger.info 'Check if CatalogSource is installed'
-  if oc get catalogsource "$OPERATOR" -n "$OLM_NAMESPACE" >/dev/null 2>&1; then
-    logger.success 'CatalogSource is already installed.'
-    return 0
-  fi
-  install_catalogsource
-}
-
 function install_catalogsource {
   logger.info "Installing CatalogSource"
+  local rootdir="$(dirname "$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")")"
 
+  # Add a user that is allowed to pull images from the registry.
+  pull_user="puller"
+  add_user "$pull_user" "puller"
+  oc -n "$OLM_NAMESPACE" policy add-role-to-user registry-viewer "$pull_user"
+  token=$(oc --config=$pull_user.kubeconfig whoami -t)
 
-  local rootdir="$(dirname "$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")")"
-  ${rootdir}/hack/catalog.sh |\
-     sed -e "s|deploy/resources/kourier/kourier-latest.yaml|deploy/resources/kourier/kourier-latest-debug.yaml|g" |\
-     oc apply -n "$OLM_NAMESPACE" -f - || return 1
+  csv="${rootdir}/olm-catalog/serverless-operator/manifests/serverless-operator.clusterserviceversion.yaml"
+  # Create a backup of the CSV so we don't pollute the repository.
+  cp "$csv" "${rootdir}/bkp.yaml"
+
+  if [ -n "$OPENSHIFT_CI" ]; then
+    sed -i "s,image: .*:knative-operator,image: ${IMAGE_FORMAT//\$\{component\}/knative-operator}," "$csv"
+    sed -i "s,image: .*:knative-openshift-ingress,image: ${IMAGE_FORMAT//\$\{component\}/knative-openshift-ingress}," "$csv"
+  elif [ -n "$DOCKER_REPO_OVERRIDE" ]; then
+    sed -i "s,image: .*:knative-operator,image: ${DOCKER_REPO_OVERRIDE}/knative-operator," "$csv"
+    sed -i "s,image: .*:knative-openshift-ingress,image: ${DOCKER_REPO_OVERRIDE}/knative-openshift-ingress," "$csv"
+  fi
+
+  cat "$csv"
+
+  # Build the bundle image in the cluster-internal registry.
+  oc -n "$OLM_NAMESPACE" new-build --binary --strategy=docker --name serverless-bundle
+  oc -n "$OLM_NAMESPACE" start-build serverless-bundle --from-dir olm-catalog/serverless-operator -F
+
+  # Undo potential changes to the CSV to not pollute the repository.
+  mv "${rootdir}/bkp.yaml" "$csv"
+
+  # HACK: Allow to run the index pod as root so it can create the necessary directories.
+  oc -n "$OLM_NAMESPACE" adm policy add-scc-to-user anyuid -z default
```

Comment:
Why we need this? Couldn't it operate on `/home` or `/tmp` subdirs?

Reply:
To be frank, I didn't have a ton of time to dig into why this image is behaving like this specifically. I"d personally go with this an fix in a followup to `master`. Not the end of the world for CI anyway imo.